### PR TITLE
Fix crash when call hash.

### PR DESCRIPTION
**Pull request by CPWstatic**
close#653

### CODE REVIEW

**Comment by CPWstatic** [comment]
Jenkins go.

File: src/common/filter/FunctionManager.cpp [hunk 1]
```diff
@@ -249,8 +249,26 @@ FunctionManager::FunctionManager() {
...
+                case 0: {
+                    auto v = Expression::asInt(args[0]);
+                    return static_cast<int64_t>(std::hash<int64_t>()(v));
+                }
+                case 1: {
+                    auto v = Expression::asDouble(args[0]);
+                    return static_cast<int64_t>(std::hash<double>()(v));
+                }
+                case 2: {
+                    auto v = Expression::asBool(args[0]);
```
Comment (laura-ding) [inline src/common/filter/FunctionManager.cpp]: is it necessary to hash a bool?

Comment (whitewum) [reply src/common/filter/FunctionManager.cpp]: > is it necessary to hash a bool?

maybe some code will do that

File: src/graph/test/YieldTest.cpp [hunk 2]
```diff
@@ -60,5 +60,60 @@ TEST_F(YieldTest, Basic) {
...
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        std::vector<std::tuple<int64_t>> expected{
+            std::hash<double>{}(123.0)
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query = "YIELD hash(!0)";
```
Comment (whitewum) [inline src/graph/test/YieldTest.cpp]: what if console input some logcial operation?
TRUE AND FALSE OR TRUE

File: src/graph/test/YieldTest.cpp [hunk 3]
```diff
@@ -60,5 +60,60 @@ TEST_F(YieldTest, Basic) {
...
+        auto code = client->execute(query, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+        std::vector<std::tuple<int64_t>> expected{
+            std::hash<int64_t>{}(123 + 456)
+        };
+        ASSERT_TRUE(verifyResult(resp, expected));
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string query = "YIELD hash(123.0)";
```
Comment (whitewum) [inline src/graph/test/YieldTest.cpp]: try hash(hash(wahaha))

**Pull request merged**